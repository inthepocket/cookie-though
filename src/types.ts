export type CookiePreference = Pick<Policy, 'id'> & {
  isEnabled: boolean;
};

export interface CookiePreferences {
  isCustomised: boolean;
  cookieOptions: CookiePreference[];
}

export type CookieOption = Policy & {
  isEnabled: boolean;
};

export interface CookieOptions extends CookiePreferences {
  cookieOptions: CookieOption[];
}
export interface Policy {
  id: string;
  label: string;
  description: string;
  category: 'essential' | 'social' | 'personalisation' | 'marketing' | 'statistics' | 'functional';
}

export interface Config {
  policies: Policy[];
  essentialLabel?: string;
  permissionLabels: {
    accept: string;
    acceptAll: string;
    decline: string;
  };
  cookiePreferenceKey?: string;
  header: {
    title?: string;
    subTitle?: string;
    description: string;
  };
  customizeLabel: string;
  cookiePolicy?: {
    url: string;
    label: string;
  };
}
