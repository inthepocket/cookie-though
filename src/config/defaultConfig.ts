import { Policy } from '../types';

export const defaultPolicies: Policy[] = [
  {
    id: 'essential',
    label: 'Strictly necessary cookies',
    description: 'These cookies are required to run the site.',
    category: 'essential',
  },
  {
    id: 'functional',
    label: 'Functional cookies',
    description: 'We’ll remember the basics such as language.',
    category: 'functional',
  },
  {
    id: 'statistics',
    label: 'Statistics',
    description: 'We’ll know where we should improve your experience.',
    category: 'statistics',
  },
  {
    id: 'marketing',
    label: 'Marketing',
    description: "We'll only show you ads you're interested in.",
    category: 'marketing',
  },
  {
    id: 'social',
    label: 'Social',
    description: 'This allows us to track your social activity',
    category: 'social',
  },
  {
    id: 'personalisation',
    label: 'Personalisation',
    description: 'We’ll only show you content that interests you.',
    category: 'personalisation',
  },
];

const defaultConfig = {
  policies: defaultPolicies,
  essentialLabel: 'Always on',
  header: {
    subTitle: "You're probably fed up with these banners...",
    title: 'cookie though?',
    description:
      "Everybody wants to show his best side - and so do we. That's why we use cookies to guarantee you a better experience.",
  },
  permissionLabels: {
    accept: 'Accept',
    acceptAll: 'Accept all',
    decline: 'Decline',
  },
  customizeLabel: 'Customize',
};

export default defaultConfig;
