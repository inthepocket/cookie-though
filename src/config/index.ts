import type { OnPreferencesChanged } from './../context/onPreferencesChanged/index';

export const COOKIE_PREFERENCES_KEY = 'cookie-preferences';

/**
 * The category to which a policy belongs to.
 */
export type Category =
  | 'essential'
  | 'social'
  | 'personalisation'
  | 'marketing'
  | 'statistics'
  | 'functional';

export interface Policy {
  /**
   * The id that is used to save the user's preferences in the cookie.
   *
   * Example: `functional`
   */
  id: string;
  /**
   * The label of the policy checkbox shown to the user in bold.
   *
   * Example: `Functional Cookies`
   */
  label: string;
  /**
   * The label of the policy checkbox shown to the user.
   *
   * Example: `We'll remember the basics such as language.`
   */
  description: string;
  /**
   * The category to which this policy belongs to.
   */
  category: Category;
}

export interface Config {
  /**
   * Announces the presence of a cookie banner to a user using a screenreader.
   *
   * Default: `Cookie Banner`.
   */
  ariaLabel?: string;
  /**
   * Announces a group of a cookie options to a user using a screenreader.
   *
   * Default: `Cookie options`.
   */
  optionsAriaLabel?: string;
  /**
   * An array of policies that are shown to the user.
   *
   * Default: a list of all possible policies.
   */
  policies: Policy[];
  /**
   * The label that is displayed instead of the checkbox for essential cookies.
   * Seeing as essential cookies can't be turned off, we display a label instead.
   *
   * Default:`Always on`.
   */
  essentialLabel?: string;
  /**
   * An object which contains the labels that are displayed to the user to accept (all) or decline cookies.
   */
  permissionLabels?: {
    accept?: string;
    acceptAll?: string;
    decline?: string;
  };
  /**
   * The key that is used for saving the cookie with the user's preferences.
   *
   * Default: `cookie-preferences`.
   */
  cookiePreferencesKey?: string;
  /**
   * An object which contains the labels that are displayed in the header.
   * In order for Cookie Though not to render one or more of these labels, pass `null` as the value.
   */
  header?: {
    /** Default: `You're probably fed up with these banners...` */
    intro?: string | null;
    /** Default: `cookie though?` */
    title?: string | null;
    /** Default: `Everybody wants to show his best side - and so do we. That's why we use cookies to guarantee you a better experience.` */
    description?: string | null;
  };
  /**
   * The label for the customize <summary> tag which opens and closes the cookie options.
   *
   * Default: `Customize`
   */
  customizeLabel?: string;
  /**
   * An object which contains a url and label which allow the user to read the full cookie policy.
   */
  cookiePolicy?: {
    url: string;
    label: string;
  };
  /**
   * A function which will be called whenever the preferences of the user change.
   */
  onPreferencesChanged?: OnPreferencesChanged;
}

const defaultPolicies: Policy[] = [
  {
    id: 'essential',
    label: 'Strictly necessary cookies',
    description: 'These cookies are required to run the site.',
    category: 'essential',
  },
  {
    id: 'functional',
    label: 'Functional cookies',
    description: "We'll remember the basics such as language.",
    category: 'functional',
  },
  {
    id: 'statistics',
    label: 'Statistics',
    description: "We'll know where we should improve your experience.",
    category: 'statistics',
  },
  {
    id: 'marketing',
    label: 'Marketing',
    description: "We'll only show you ads you're interested in.",
    category: 'marketing',
  },
  {
    id: 'social',
    label: 'Social',
    description: 'This allows us to track your social activity',
    category: 'social',
  },
  {
    id: 'personalisation',
    label: 'Personalisation',
    description: "We'll only show you content that interests you.",
    category: 'personalisation',
  },
];

const defaultConfig = {
  ariaLabel: 'Cookie banner',
  optionsAriaLabel: 'Cookie options',
  policies: defaultPolicies,
  essentialLabel: 'Always on',
  header: {
    intro: "You're probably fed up with these banners...",
    title: 'cookie though?',
    description:
      "Everybody wants to show his best side - and so do we. That's why we use cookies to guarantee you a better experience.",
  },
  permissionLabels: {
    accept: 'Accept',
    acceptAll: 'Accept all',
    decline: 'Decline',
  },
  customizeLabel: 'Customize',
  cookiePreferencesKey: COOKIE_PREFERENCES_KEY,
};

export default defaultConfig;
