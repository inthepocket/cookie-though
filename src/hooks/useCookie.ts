import { CookiePreference, CookiePreferences } from '../types';
import { EventEmitter } from 'events';

export const COOKIE_PREFERENCES_KEY = 'cookie-preferences';
export const COOKIE_PREFERENCES_CHANGED_EVENT = 'preferences_changed';

interface Options {
  cookieOptions: CookiePreference[];
  cookiePreferenceKey?: string;
  domain?: string;
  ee?: EventEmitter;
}

export const getCookie = (cookieKey: string) => {
  const rawCookies = decodeURIComponent(document.cookie).split(';');
  const currentPreferences = rawCookies.reduce<string[] | undefined>(
    (cookiePreferences, cookie) => {
      if (cookiePreferences) return cookiePreferences;

      const [key, value] = cookie.split('=');
      if (key.trim() === cookieKey) return value.split('|');
    },
    undefined,
  );

  if (!currentPreferences) {
    return undefined;
  }

  /*
    Clear old cookie preferences
  */
  if (currentPreferences[0].includes('cookieOptions')) {
    document.cookie = `${cookieKey}= ; expires= Thu, 01 Jan 1970 00:00:00 GMT; path=/`;
    return undefined;
  }

  return {
    isCustomised: true,
    cookieOptions: currentPreferences.map(preference => {
      const [id, isEnabled] = preference.split(':');
      return { id, isEnabled: !!+isEnabled };
    }),
  };
};

export const getNextYear = () => {
  const now = new Date();
  return new Date(now.setFullYear(now.getFullYear() + 1)).toUTCString();
};

const policiesHaveChanged = (
  policyIds: CookiePreference['id'][],
  preferenceIds: CookiePreference['id'][],
) => {
  return policyIds.some(id => !policyIds.includes(id)) || policyIds.length !== preferenceIds.length;
};

const mergePolicies = (
  policies: CookiePreference[],
  preferences: CookiePreference[],
): CookiePreference[] => {
  return policies.map(policy => {
    const isEnabled = preferences.reduce((isPreferenceEnabled, preference) => {
      return preference.id === policy.id ? preference.isEnabled : isPreferenceEnabled;
    }, false);

    return { ...policy, isEnabled };
  });
};

export const getCookiePreferences = (options: CookiePreference[], cookiePreferenceKey: string) => {
  const defaultPreferences: CookiePreferences = {
    cookieOptions: options,
    isCustomised: false,
  };
  const cookiePreferences = getCookie(cookiePreferenceKey);

  if (!cookiePreferences) return defaultPreferences;

  const policyIds = defaultPreferences.cookieOptions.map(cookieOption => cookieOption.id);
  const preferenceIds = cookiePreferences.cookieOptions.map(cookieOption => cookieOption.id);
  if (!policiesHaveChanged(policyIds, preferenceIds)) return cookiePreferences;

  const cookieOptions = mergePolicies(
    defaultPreferences.cookieOptions,
    cookiePreferences.cookieOptions,
  );

  return { cookieOptions, isCustomised: false };
};

export const formatToCookie = (cookiePreferences: CookiePreference[]) => {
  return cookiePreferences.reduce((formattedCookieValue, preference, i) => {
    return `${formattedCookieValue}${i !== 0 ? '|' : ''}${preference.id}:${
      preference.isEnabled ? '1' : '0'
    }`;
  }, '');
};

const useCookie = ({
  cookieOptions,
  domain,
  cookiePreferenceKey = COOKIE_PREFERENCES_KEY,
  ee,
}: Options) => {
  const setCookiePreferences = (cookiePreferences: CookiePreferences) => {
    const expires = getNextYear();
    document.cookie = `${cookiePreferenceKey}=${formatToCookie(
      cookiePreferences.cookieOptions,
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    )}; expires=${expires}; path=/; SameSite=Strict;${domain ? `Domain=${domain}` : ''}`;
    if (ee) {
      ee.emit(COOKIE_PREFERENCES_CHANGED_EVENT, cookiePreferences);
    }
    return cookiePreferences;
  };

  return {
    getCookiePreferences: () => getCookiePreferences(cookieOptions, cookiePreferenceKey),
    setCookiePreferences,
  };
};

export default useCookie;
