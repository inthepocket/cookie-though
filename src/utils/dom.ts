export const setVisible = (value: boolean) => {
  const container = document.querySelector<HTMLElement>('.cookie-though')!;
  if (value) {
    container.style.display = 'block';
    container.setAttribute('aria-hidden', 'false');
    return container.classList.add('visible');
  }
  container.setAttribute('aria-hidden', 'true');
  container.classList.remove('visible');
  setTimeout(() => (container.style.display = 'none'), 250);
};
