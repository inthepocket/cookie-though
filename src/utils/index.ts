export * from './category';
export * from './dom';
