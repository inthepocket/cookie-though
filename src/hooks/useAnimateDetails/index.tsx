import { RefObject } from 'preact';
import { useCallback, useEffect, useLayoutEffect, useState } from 'preact/hooks';

import { getContainer } from '../../utils/container';
import usePrefersReducedMotion from '../usePrefersReducedMotion';
import useTransitionDuration from '../useTransitionDuration';

const CLOSED_SIZE = '0px';
const easing = 'ease-in-out';
let wantedHeight = 0;
let contentMarginBottom = 0;

export const getFloat = (element: HTMLElement, key: 'height' | 'marginBottom') => {
  const value = window.getComputedStyle(element)[key];

  if (value === 'auto' || value === '') return 0;
  return parseFloat(value);
};

const getOffset = (element: HTMLElement) => {
  return getFloat(element, 'height') + getFloat(element, 'marginBottom');
};

const getOpenHeight = (details: HTMLDetailsElement) => {
  const wantedOffset = wantedHeight + contentMarginBottom;
  const header = details.previousElementSibling as HTMLElement;
  const summary = details.querySelector('summary')!;
  const acceptance = details.nextElementSibling as HTMLDivElement;

  const headerOffset = getOffset(header);
  const summaryOffset = getOffset(summary);
  const acceptanceOffset = getOffset(acceptance);
  const bannerOffset = parseInt(getComputedStyle(getContainer()!).paddingBottom, 10) * 4;

  const maxOffset =
    window.innerHeight - headerOffset - summaryOffset - acceptanceOffset - bannerOffset;
  const maxHeight = maxOffset - bannerOffset / 4;
  const hasOverflow = wantedOffset > maxOffset;
  const openHeight = `${hasOverflow ? maxHeight : wantedHeight}px`;

  return { hasOverflow, openHeight, maxHeight: `${maxHeight}px` };
};

type AnimateParams = {
  details: HTMLDetailsElement;
  isOpen: boolean;
};

const useAnimateDetails = (detailsRef: RefObject<HTMLDetailsElement>) => {
  const [details, setDetails] = useState<HTMLDetailsElement | null>(null);
  const [isOpen, setIsOpen] = useState(false);
  const prefersReducedMotion = usePrefersReducedMotion();
  const duration = useTransitionDuration();

  useEffect(() => {
    if (!detailsRef.current) return;

    setDetails(detailsRef.current);
  }, [details, detailsRef]);

  const toggleDetails = useCallback(() => {
    const content = details!.querySelector<HTMLDivElement>('.content')!;
    const { maxHeight } = getOpenHeight(details!);
    content.style.maxHeight = maxHeight;
    content.style.overflowY = 'scroll';
    setIsOpen((prevState) => !prevState);
  }, [details]);

  const animateDetails = useCallback(
    (e: MouseEvent) => {
      e.preventDefault();
      const content = details!.querySelector<HTMLDivElement>('.content')!;

      const animate = ({ details, isOpen }: AnimateParams) => {
        const content = details.querySelector<HTMLDivElement>('.content')!;
        const { openHeight, hasOverflow } = getOpenHeight(details);
        const height = isOpen ? [CLOSED_SIZE, openHeight] : [openHeight, CLOSED_SIZE];

        const animation = content.animate({ height }, { duration, easing });
        animation.onfinish = () => {
          if (!isOpen) {
            details.open = isOpen;
            setIsOpen(isOpen);
            return content.removeAttribute('style');
          }

          content.style.height = openHeight;
          content.style.marginBottom = `${contentMarginBottom}px`;

          return hasOverflow
            ? (content.style.overflowY = 'scroll')
            : content.style.removeProperty('overflowY');
        };
      };

      const closeDetails = () => {
        return window.requestAnimationFrame(() => animate({ details: details!, isOpen: false }));
      };

      const openDetails = () => {
        const content = details!.querySelector<HTMLDivElement>('.content')!;

        details!.open = true;
        setIsOpen(true);
        wantedHeight = getFloat(content, 'height');
        contentMarginBottom = getFloat(content, 'marginBottom');
        content.style.height = CLOSED_SIZE;
        content.style.marginBottom = CLOSED_SIZE;
        window.requestAnimationFrame(() => animate({ details: details!, isOpen: true }));
      };

      content.style.overflow = 'hidden';
      details!.open ? closeDetails() : openDetails();
    },
    [details, duration],
  );

  useLayoutEffect(() => {
    if (!details) return;
    const summary = details.querySelector('summary')!;

    if (prefersReducedMotion) {
      summary.removeEventListener('click', animateDetails);
      return summary.addEventListener('click', toggleDetails);
    }

    summary.removeEventListener('click', toggleDetails);
    summary.addEventListener('click', animateDetails);
  }, [animateDetails, details, duration, prefersReducedMotion, toggleDetails]);

  return { isOpen };
};

export default useAnimateDetails;
