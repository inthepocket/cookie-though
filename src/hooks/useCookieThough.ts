import { useEffect, useState } from 'preact/hooks';
import { CookiePreferences, Policy } from '../types';

function isEnabled(preferences: CookiePreferences, microPolicy: Policy['id']): boolean {
  const option = preferences.cookieOptions.find(x => x.id === microPolicy);
  if (!option) {
    return false;
  }
  return option.isEnabled;
}

export default function useCookieThough(
  listen: (cb: (preferences: CookiePreferences) => void) => void,
  getCookiePreferences: () => CookiePreferences,
  micropolicy: Policy['id'],
) {
  const [hasConsent, setConsentState] = useState<boolean>(false);

  useEffect(() => {
    const prefs = getCookiePreferences();
    setConsentState(isEnabled(prefs, micropolicy));
    listen(preferences => {
      setConsentState(isEnabled(preferences, micropolicy));
    });
    /* eslint-enable no-unused-expressions */
  }, [micropolicy, listen, getCookiePreferences]);

  return !!hasConsent;
}
