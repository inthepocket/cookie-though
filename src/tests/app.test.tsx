import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { englishMockPolicies } from './__mocks__/policies';
import { COOKIE_PREFERENCES_KEY, formatToCookie } from '../hooks/useCookie';
import { englishMockConfig } from './__mocks__/config';
import App, {
  CONTAINER_BOTTOMS,
  CONTAINER_WIDTHS,
  MOBILE_CONTAINER_BOTTOMS,
} from '../components/app';
import clearCookies from './helpers/clearCookies';
import { CookiePreferences } from '../types';

jest.mock('../utils/dom', () => ({
  setVisible: jest.fn().mockImplementation(() => 'You have called setVisible'),
}));
import { setVisible } from '../utils/dom';
const { description } = englishMockConfig.header;

describe('Cookie Though', () => {
  let container: HTMLDivElement;
  beforeAll(() => {
    Object.defineProperty(window, 'getComputedStyle', {
      value: () => ({ fontSize: '12px', height: '0px', bottom: '0px' }),
    });
  });

  beforeEach(() => {
    const manageCookiesElement = document.createElement('button');
    manageCookiesElement.id = 'manage-cookie-though';
    document.body.append(manageCookiesElement);

    container = document.createElement('div');
    container.className = 'cookie-though';
    container.attachShadow({ mode: 'open' });

    const textDiv = document.createElement('div');
    textDiv.className = 'ct-banner-explanation';
    container.shadowRoot!.appendChild(textDiv);
    document.body.appendChild(container);
  });

  afterEach(() => {
    clearCookies();
    document.getElementsByTagName('html')[0].innerHTML = '';
  });

  describe('without cookie preferences stored in a cookie', () => {
    it('should show the cookie wall', () => {
      mount(
        <App
          customizeLabel="customize"
          header={{ description }}
          policies={englishMockConfig.policies}
          permissionLabels={englishMockConfig.permissionLabels}
        />,
        { attachTo: container },
      );

      expect(setVisible).toBeCalled();
    });
  });

  describe('with user preferences stored in a cookie', () => {
    const DEFAULT_COOKIE_PREFERENCES: CookiePreferences = {
      isCustomised: false,
      cookieOptions: englishMockConfig.policies.map(policy => ({ ...policy, isEnabled: false })),
    };

    it('should not show the cookie wall', () => {
      document.cookie = `${COOKIE_PREFERENCES_KEY}=${formatToCookie(
        DEFAULT_COOKIE_PREFERENCES.cookieOptions,
      )}`;
      mount(
        <div className="cookie-though">
          <App
            customizeLabel="customize"
            header={{ description }}
            policies={englishMockPolicies}
            permissionLabels={englishMockConfig.permissionLabels}
          />
        </div>,
        { attachTo: document.body },
      );
    });

    it("should show the cookie wall if the cookie preferences aren't customised", () => {
      jest.mock('../utils/dom', () => ({
        setVisible: jest.fn().mockImplementation(() => 'You have called setVisible'),
      }));
      mount(
        <div className="cookie-though">
          <App
            customizeLabel="customize"
            header={{ description }}
            policies={englishMockPolicies}
            permissionLabels={englishMockConfig.permissionLabels}
          />
        </div>,
        { attachTo: document.body },
      );

      expect(setVisible).toBeCalled();
    });
  });

  it('should render properly', () => {
    const wrapper = shallow(
      <body>
        <button id="manage-cookie-though"></button>
        <div className="cookie-though">
          <App
            customizeLabel="customize"
            header={{ description }}
            policies={englishMockPolicies}
            permissionLabels={englishMockConfig.permissionLabels}
          />
        </div>
      </body>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('if the user has a different browser font setting', () => {
    const renderApp = (fontSize: string) => {
      container.style.fontSize = fontSize;
      return mount(
        <App
          customizeLabel="customize"
          header={{ description }}
          policies={englishMockPolicies}
          permissionLabels={englishMockConfig.permissionLabels}
        />,
        { attachTo: container },
      );
    };

    const mockGetComputedStyle = (size: number) => {
      Object.defineProperty(window, 'getComputedStyle', {
        value: () => ({ fontSize: `${size}px`, height: '0px', bottom: '0px' }),
      });
    };

    it('should adjust the width and bottom of the container based on the font', () => {
      const fontSizes = [13.5, 15, 17, 19, 23];
      fontSizes.forEach((fontSize, i) => {
        mockGetComputedStyle(fontSize);
        renderApp(`${fontSize}px`);
        const container = document.querySelector('.cookie-though') as HTMLElement;
        switch (i) {
          case 2:
            expect(container.style.width).toEqual(CONTAINER_WIDTHS[2]);
            expect(container.style.bottom).toEqual(CONTAINER_BOTTOMS[1]);
            break;
          case 3:
            expect(container.style.width).toEqual(CONTAINER_WIDTHS[3]);
            expect(container.style.bottom).toEqual(CONTAINER_BOTTOMS[2]);
            break;
          case 4:
            expect(container.style.width).toEqual(CONTAINER_WIDTHS[3]);
            expect(container.style.bottom).toEqual(CONTAINER_BOTTOMS[3]);
            break;
          default:
            expect(container.style.width).toEqual(CONTAINER_WIDTHS[i]);
            expect(container.style.bottom).toEqual(CONTAINER_BOTTOMS[i]);
        }
      });
    });

    it('should not adjust the width of the container if the width is smaller than the breakpoint', () => {
      global.innerWidth = 375;
      const fontSizes = [13.5, 15, 17, 19, 23];
      fontSizes.forEach(fontSize => {
        mockGetComputedStyle(fontSize);
        renderApp(`${fontSize}px`);
        const container = document.querySelector('.cookie-though') as HTMLElement;
        expect(container.style.width).toBe('');
      });
    });

    it('should set the bottom attribute if the width is smaller than the breakpoint and the font size is large', () => {
      const fontSizes = [17, 19, 23];
      global.innerWidth = 375;
      fontSizes.forEach((fontSize, i) => {
        mockGetComputedStyle(fontSize);
        renderApp(`${fontSize}px`);
        const container = document.querySelector('.cookie-though') as HTMLElement;
        expect(container.style.bottom).toEqual(MOBILE_CONTAINER_BOTTOMS[i]);
      });
    });
  });
});
