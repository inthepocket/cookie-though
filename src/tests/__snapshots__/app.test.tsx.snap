// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Though should render properly 1`] = `
<body>
  <button
    id="manage-cookie-though"
  />
  <div
    className="cookie-though"
  >
    <App
      customizeLabel="customize"
      header={
        Object {
          "description": "Everybody wants to show his best side - and so do we. That's why we use cookies to guarantee you a better experience.",
        }
      }
      permissionLabels={
        Object {
          "accept": "Accept",
          "acceptAll": "Accept all",
          "decline": "Decline",
        }
      }
      policies={
        Array [
          Object {
            "category": "essential",
            "description": "These cookies are required to run the site.",
            "id": "essential",
            "label": "Strictly necessary cookies",
          },
          Object {
            "category": "functional",
            "description": "We’ll remember the basics such as language.",
            "id": "functional",
            "label": "Functional cookies",
          },
          Object {
            "category": "statistics",
            "description": "We’ll know where we should improve your experience.",
            "id": "statistics",
            "label": "Statistics",
          },
          Object {
            "category": "marketing",
            "description": "We’ll only show you content that interests you.",
            "id": "marketing",
            "label": "Marketing",
          },
        ]
      }
    />
  </div>
</body>
`;
